### ISSUE TITLE
 Import OutOfMemoryError

### ISSUE THREAD

**Issue by mongoose54** [neo4j/neo4j#9264]
I am facing a java heap space issue when importing a 3.5M nodes with 500K relationships.

**System Specs:
- Neo4j version:  3.1.3
- Operating system: Ubuntu 14.10
- API/Driver: Neo4j Import**

This is the command:
`./bin/neo4j-import --into graph.db --nodes nodes.csv --relationships rels.csv --delimiter TAB --stacktrace
`

And this the error:

```
Exception in thread "main" java.lang.OutOfMemoryError: Java heap space: Not enough memory available for allocating 4.07 GB, tried [OFF_HEAP, HEAP]: Not enough memory available for allocating 40.70 GB, tried [OFF_HEAP, HEAP, CHUNKED_FIXED_SIZE]
	at org.neo4j.unsafe.impl.batchimport.cache.HeapLongArray.<init>(HeapLongArray.java:36)
	at org.neo4j.unsafe.impl.batchimport.cache.NumberArrayFactory$1.newLongArray(NumberArrayFactory.java:153)
	at org.neo4j.unsafe.impl.batchimport.cache.NumberArrayFactory$Auto.newLongArray(NumberArrayFactory.java:221)
	at org.neo4j.unsafe.impl.batchimport.cache.DynamicLongArray.addChunk(DynamicLongArray.java:66)
	at org.neo4j.unsafe.impl.batchimport.cache.DynamicLongArray.addChunk(DynamicLongArray.java:28)
	at org.neo4j.unsafe.impl.batchimport.cache.DynamicNumberArray.synchronizedAddChunk(DynamicNumberArray.java:98)
	at org.neo4j.unsafe.impl.batchimport.cache.DynamicNumberArray.at(DynamicNumberArray.java:84)
	at org.neo4j.unsafe.impl.batchimport.cache.NumberArrayFactory$3.newLongArray(NumberArrayFactory.java:290)
	at org.neo4j.unsafe.impl.batchimport.cache.NumberArrayFactory$Auto.newLongArray(NumberArrayFactory.java:221)
	at org.neo4j.unsafe.impl.batchimport.cache.NumberArrayFactory.newLongArray(NumberArrayFactory.java:72)
	at org.neo4j.unsafe.impl.batchimport.RelationshipCountsProcessor.<init>(RelationshipCountsProcessor.java:63)
	at org.neo4j.unsafe.impl.batchimport.ProcessRelationshipCountsDataStep.processor(ProcessRelationshipCountsDataStep.java:73)
	at org.neo4j.unsafe.impl.batchimport.ProcessRelationshipCountsDataStep.process(ProcessRelationshipCountsDataStep.java:60)
	at org.neo4j.unsafe.impl.batchimport.ProcessRelationshipCountsDataStep.process(ProcessRelationshipCountsDataStep.java:36)
	at org.neo4j.unsafe.impl.batchimport.staging.ProcessorStep.lambda$receive$2(ProcessorStep.java:94)
	at org.neo4j.unsafe.impl.batchimport.staging.ProcessorStep$$Lambda$78/137690261.run(Unknown Source)
	at org.neo4j.unsafe.impl.batchimport.executor.DynamicTaskExecutor$Processor.run(DynamicTaskExecutor.java:241)

```

**Comment by ketank-new** [comment]
@mongoose54 : Try exporting 

export MAVEN_OPTS="-Xmx1024m -XX:MaxPermSize=512m"
export JVM_ARGS="-Xms1024m -Xmx1024m"
export JAVA_OPTS="-Xms128m -Xmx512m"

and give another try , let me know if your heap issues gets resolved with this

**Comment by chrisvest** [comment]
This should be fixed in a recent-ish 3.2 with #9670 and #9762.
